sdk/core/azure-core/src/main/java/com/azure/core/implementation/SwaggerInterfaceParser.java

```diff
@@ -20,7 +20,7 @@
 public class SwaggerInterfaceParser {
     private final String host;
     private final String serviceName;
-    private final Map<Method, SwaggerMethodParser> methodParsers = new HashMap<>();
+    private static final Map<Method, SwaggerMethodParser> METHOD_PARSERS = new ConcurrentHashMap<>();
```

Comment:
Is this still necessary to have after making the change to `getBlockBlobAsyncClient()`?

Reply:
Let's keep it as it's going to improve performance.

Reply:
Yup, that call is what triggers this code path, we may have reduced the number of calls but this was the hot spot.